pkg/operator/operator.go

```diff
@@ -133,6 +137,43 @@ func New(ctx context.Context, config operatorconfig.Config, kubeConfig *rest.Con
 		return nil, fmt.Errorf("failed to create status controller: %v", err)
 	}
 
+	// Set up the DNSNameResolver controller. This controller is unmanaged by
+	// the manager. The reason why the controller is unmanaged is so that it
+	// doesn't get automatically started when the operator starts; we only
+	// want the controller to start if we need it. The controller re-resolves
+	// the DNS names which get added to the status of the DNSNameResolver objects.
+	// It also removes IP addresses of DNS names, whose TTLs have expired,
+	// from the status. The enable_dnsnameresolver controller starts it and the
+	// caches, namely dnsNameResolverCache and corednsEndpointsSliceCache.
+	// TODO: Make dnsnameresolver controller managed and remove dependence
+	// on enablednsnameresolver controller to start it and the related caches.
+	dnsNameResolverController, dnsNameResolverControllerCaches, err := dnsnameresolver.NewUnmanaged(operatorManager, dnsnameresolver.Config{
+		OperandNamespace: operatorcontroller.DefaultOperandNamespace,
+		ServiceName: operatorcontroller.DNSServiceName(&operatorv1.DNS{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: operatorcontroller.DefaultDNSController,
+			},
+		}).Name,
+		DNSPort:                  fmt.Sprint(manifests.DNSDaemonSet().Spec.Template.Spec.Containers[0].Ports[0].ContainerPort),
+		DNSNameResolverNamespace: operatorcontroller.DefaultDNSNameResolverNamespace,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create dnsnameresolver controller: %v", err)
+	}
+
+	// Set up the enable_dnsnameresolver controller. This controller starts the
+	// dnsnameresolver controller and the dependent caches, namely dnsNameResolverCache
+	// and corednsEndpointsSliceCache.
+	if _, err := enablednsnameresolver.New(operatorManager, enablednsnameresolver.Config{
+		DNSNameResolverEnabled: dnsNameResolverEnabled,
+		DependentCaches:        dnsNameResolverControllerCaches,
+		DependentControllers: []controller.Controller{
+			dnsNameResolverController,
+		},
+	}); err != nil {
+		return nil, fmt.Errorf("failed to create enable_dnsnameresolver controller: %w", err)
```

Comment:
nit: we typically use `%v` unless there is a specific reason to use `%w`.

Reply:
We should be using `%w` in all new code that uses `Errorf` to wrap errors.  The only reason we have `%v` in some places where we could use `%w` is because we have a lot of pre-Go 1.13 code ([`%w` was introduced in Go 1.13](https://go.dev/doc/go1.13#error_wrapping)).  

That said, this is definitely something that can be deferred to a post-branch-cut followup.